src/thor/costmatrix.cc

```diff
@@ -326,22 +331,40 @@ void CostMatrix::Initialize(
                     return limits.max_up_transitions == kUnlimitedTransitions;
                   });
 
-  // Add initial sources status
-  for (const auto exp_dir : {MATRIX_FORW, MATRIX_REV}) {
-    const auto count = locs_count_[exp_dir];
-    locs_status_[exp_dir].reserve(count);
-    hierarchy_limits_[exp_dir].resize(count);
-    adjacency_[exp_dir].resize(count);
-    edgestatus_[exp_dir].resize(count);
-    edgelabel_[exp_dir].resize(count);
+  const uint32_t bucketsize = costing_->UnitSize();
+  const float range = kBucketCount * bucketsize;
```

Comment:
changed this to be the bucket size as in bidirectional_astar.cc. this might change again, see https://github.com/valhalla/valhalla/issues/3605#issuecomment-2028938018